src/scripts/main.js

```diff
@@ -1,3 +1,391 @@
-'use strict';
+import 'regenerator-runtime/runtime';
 
-// write your code here
+// Cell
+
+const score = document.querySelector('.game_score');
+let scoreValue = 0;
+
+class Cell {
+  constructor(gridElement, x, y) {
+    const cell = document.createElement('div');
+
+    cell.classList.add('field_cell');
+    gridElement.append(cell);
+    this.x = x;
+    this.y = y;
+  }
+
+  restartScore() {
+    scoreValue = 0;
+    score.innerHTML = scoreValue;
+  }
+
+  linkTile(tile) {
+    tile.setXY(this.x, this.y);
+    this.linkedTile = tile;
+  }
+
+  unlinkTile() {
+    this.linkedTile = null;
+  }
+
+  isEmpty() {
+    return !this.linkedTile;
+  }
+
+  linkTileForMerge(tile) {
+    tile.setXY(this.x, this.y);
+    this.linkedTileForMerge = tile;
+  }
+
+  unlinkTileForMerge() {
+    this.linkedTileForMerge = null;
+  }
+
+  hasTileForMerge() {
+    return !!this.linkedTileForMerge;
+  }
+
+  canAccept(newTile) {
+    return this.isEmpty()
+    || (!this.hasTileForMerge() && this.linkedTile.value === newTile.value);
+  }
+
+  mergeTiles() {
+    this.linkedTile.setValue(this.linkedTile.value * 2);
+    scoreValue += this.linkedTile.value;
+    score.innerHTML = scoreValue;
+    this.linkedTileForMerge.removeFromDOM();
+    this.unlinkTileForMerge();
+  }
+}
+
+// Grid
+
+const GRID_SIZE = 4;
+
+class Grid {
+  constructor(gridElement) {
+    this.cells = [];
+
+    for (let i = 0; i < GRID_SIZE * GRID_SIZE; i++) {
+      this.cells.push(
+        new Cell(gridElement, i % GRID_SIZE, Math.floor(i / GRID_SIZE))
+      );
+    }
+
+    this.cellsGroupedByColumn = this.groupCellsByColumn();
+
+    this.cellsGroupedByReversedColumn = this.cellsGroupedByColumn.map(
+      (column) => {
+        return [...column].reverse();
+      }
+    );
+    this.cellsGroupedByRow = this.groupCellsByRow();
+
+    this.cellsGroupedByReversedRow = this.cellsGroupedByRow.map(
+      (row) => {
+        return [...row].reverse();
+      }
+    );
```

Comment:
these are too many lines of code in a constructor, I suggest you to divide this code on the separate functions